sql/analyzer/tables.go

```diff
@@ -150,8 +150,10 @@ func getResolvedTable(node sql.Node) *plan.ResolvedTable {
 
 		switch n := node.(type) {
 		case *plan.ResolvedTable:
-			table = n
-			return false
+			if !plan.IsDualTable(n) {
```

Comment:
`getResolvedTable` is used in a lot of places. Do we think that none of those uses should ever see the dual table? Or is it more likely that one of the uses (wherever error happened) was underspecified? My guess is maybe the later, the place that was erroring should special handle the dual table.

Reply:
> Do we think that none of those uses should ever see the dual table?

That's what I'm thinking, yeah.

I looked through the `getResolvedTable` uses and here's the summary:
- In most cases, we get the schema from the returned table. If we're getting the schema for dualTable, that's obviously wrong, since the schema is a single name-less column and will conflict with whatever other data we have.
- In some cases the table is used to create TableAliases. This results in an empty-string alias that points to dualTable, and a table with wrong schema being used.

Based on the above, I think returning dualTable from `getResolvedTable` is a mistake: no one is expecting dualTable to be returned from `getResolvedTable`.

> Or is it more likely that one of the uses (wherever error happened) was underspecified? My guess is maybe the later, the place that was erroring should special handle the dual table.

The error is happening in [this block](https://github.com/dolthub/go-mysql-server/blob/main/sql/analyzer/resolve_tables.go#L199-L208) of `setTargetSchemas`:
```go
table := getResolvedTable(n)
if table == nil {
	return n, transform.SameTree, nil
}

var err error
n, err = t.WithTargetSchema(table.Schema())
if err != nil {
	return nil, transform.SameTree, err
}
```

We are getting the first resolved table, and then are applying that table's schema to the target schema. dualTable's schema is clearly wrong, so we should exclude it.

But we can't just handle that in `setTargetSchemas`, we'll need to either update `getResolvedTable` (to skip dualTable) or add something like `getResolvedNonDualTable`.

What do you think?